src/rocks_record_store.cpp

```diff
@@ -1190,7 +1190,6 @@ namespace mongo {
     }
 
     boost::optional<Record> RocksRecordStore::Cursor::curr() {
-        invariantRocksOK(_iterator->status());
         if (!_iterator->Valid()) {
             _eof = true;
```

Comment:
As soon as the iterator is invalid we have to check whether the status is not okay. Otherwise, we would mark the iterator as EOF and not report an error when a corruption happens.

I agree that we shouldn't call `invariant` on every `curr`, but we have to call it before we set `_eof = true;`, in the `if` condition.